recipe/meta.yaml

```
@@ -1,18 +1,17 @@
-{% set name = "arrow-c-glib" %}
-{% set version = "18.1.0" %}
+{% set version = "19.0.0" %}
 {% set version_major = version.split(".")[0] %}
 {% set version_minor = version.split(".")[1] %}
 
 package:
-  name: {{ name|lower }}
+  name: arrow-c-glib
   version: {{ version }}
 
 source:
   url: https://dist.apache.org/repos/dist/release/arrow/arrow-{{ version }}/apache-arrow-{{ version }}.tar.gz
```

Could you use `closer.lua` URL instead of `dist.apache.org` for ASF projects? `closer.lua` URL uses a suitable mirror automatically.

```suggestion
  url: https://www.apache.org/dyn/closer.lua/arrow/arrow-{{ version }}/apache-arrow-{{ version }}.tar.gz
```

See also: https://infra.apache.org/release-download-pages.html#links

---

I really dislike that this file has a different hash than the one under `/repos/dist/release`, but I've made my point about preferring fully static download urls so many times that I tire of repeating it.

---

OK, this is running into
```
  File "/opt/conda/lib/python3.12/tarfile.py", line 1842, in open
    raise ReadError(f"file could not be opened successfully:\n{error_msgs_summary}")
tarfile.ReadError: file could not be opened successfully:
- method gz: ReadError('not a gzip file')
- method bz2: ReadError('not a bzip2 file')
- method xz: ReadError('not an lzma file')
- method tar: ReadError('invalid header')
```
I won't spend more time on this, please open a PR with your preferred URL scheme. If it's green, I'll merge.

---

Thanks for taking the time @h-vetinari ! I can follow up on a different PR, from what I understand the URL you used points to an html page, if we want to download we have to use the download action, checksum is maintained:
```
$ wget https://www.apache.org/dyn/closer.lua/arrow/arrow-19.0.0/apache-arrow-19.0.0.tar.gz?action=download -O apache-arrow-19.0.0.tar.gz
...

$ sha256sum apache-arrow-19.0.0.tar.gz 
f89b93f39954740f7184735ff1e1d3b5be2640396febc872c4955274a011f56b  apache-arrow-19.0.0.tar.gz
```

---

Ah, sorry... I missed `?action=download` in my suggestion...